subsys/net/ip/net_context.c

```diff
@@ -583,6 +584,8 @@ int net_context_get(sa_family_t family, enum net_sock_type type, uint16_t proto,
 
 				contexts[i].ipv6_hop_limit = INITIAL_HOP_LIMIT;
 				contexts[i].ipv6_mcast_hop_limit = INITIAL_MCAST_HOP_LIMIT;
+				contexts[i].options.ipv6_mcast_loop =
+					IS_ENABLED(CONFIG_NET_INITIAL_IPV6_MCAST_LOOP);
```

Comment:
Doesn't this need a guard?

```suggestion
#if defined(CONFIG_NET_IPV6)
				contexts[i].options.ipv6_mcast_loop =
					IS_ENABLED(CONFIG_NET_INITIAL_IPV6_MCAST_LOOP);
#endif
```

Reply:
This code comes in the if condition of 
`if (IS_ENABLED(CONFIG_NET_IPV6) && family == AF_INET6) {`

So we do not need this.

Reply:
The `if` condition is a runtime check, but the struct member would not even be compiled in if `CONFIG_NET_IPV4` and `CONFIG_NET_IPV6` are not enabled.

Reply:
understood. I will change it.

Reply:
Added